core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -128,21 +138,83 @@
   }
 
   /**
+   * An interface extending{@link FieldIndex}, represents a unique index.
+   *
+   * @param <T> type of objects in this {@link IndexedSet}
+   */
+  public interface UniqueFieldIndex<T> extends FieldIndex<T> {}
+
+  /**
+   * An interface extending{@link FieldIndex}, represents a nonunique index.
+   *
+   * @param <T> type of objects in this {@link IndexedSet}
+   */
+  public interface NonUniqueFieldIndex<T> extends FieldIndex<T> {}
+
+  /**
    * Constructs a new {@link IndexedSet} instance with at least one field as the index.
    *
    * @param field at least one field is needed to index the set of objects
    * @param otherFields other fields to index the set
    */
   @SafeVarargs
   public IndexedSet(FieldIndex<T> field, FieldIndex<T>... otherFields) {
-    Map<FieldIndex<T>, ConcurrentHashMap<Object, ConcurrentHashSet<T>>> indexMap =
-        new HashMap<>(otherFields.length + 1);
-    indexMap.put(field, new ConcurrentHashMap<Object, ConcurrentHashSet<T>>(8, 0.95f, 8));
+    // count the numbers of two index types
+    int uniqueIndexLength = 0;
+    int nonUniqueIndexLength = 0;
+
+    if (field instanceof UniqueFieldIndex) {
+      uniqueIndexLength = 1;
+    } else {
+      nonUniqueIndexLength = 1;
+    }
+
     for (FieldIndex<T> fieldIndex : otherFields) {
-      indexMap.put(fieldIndex, new ConcurrentHashMap<Object, ConcurrentHashSet<T>>(8, 0.95f, 8));
+      if (fieldIndex instanceof UniqueFieldIndex) {
+        uniqueIndexLength++;
+      } else {
+        nonUniqueIndexLength++;
+      }
+    }
+
+    // initiation
```

Comment:
technically, initiation is a word, but I think you really mean `initialization` (and not `intialization` or `initiation`)